### PR TITLE
UTIL-1635 MyAccount - Weekendavisen - refactor login screen

### PR DESCRIPTION
**Pull request by bmxmaga**
https://berlingskemedia.atlassian.net/browse/UTIL-1635

**NOTE**: The amount of files that are changed is due to a change in the TestingPage class which is a base class for many other testing classes.
So, whenever you see a file matching this pattern: _*.page.ts_ - it means it only has a minor change, that is the `initStubs()` and `initSpies()` methods were replace by one method `init()`.

### CODE REVIEW

File: src/app/core/gigya.service.ts [hunk 1]
```diff
@@ -4,11 +4,29 @@ import { AccountInfo } from '../../../d/gigya/accounts/accounts';
...
 import { LoginEvent } from '../../../d/gigya/events/loginEvent';
 import { Response as GigyaResponse } from '../../../d/gigya/socialize/response';
+import { LoginForm } from '../login/login';
 
 declare const gigya: Gigya;
 
 @Injectable()
 export class GigyaService {
+  mapErrorMessage(response: LoginEvent): string {
+    switch (response.errorCode) {
```
Comment (bmmkoc) [inline src/app/core/gigya.service.ts]: What's the point of a switch statement with only one option? Just use an if-statement, or if you have multiple error codes to check, then put them in the top of the file, in a lookup array/object that maps error codes to error messages.

File: src/app/core/gigya.service.ts [hunk 2]
```diff
@@ -4,11 +4,29 @@ import { AccountInfo } from '../../../d/gigya/accounts/accounts';
...
+      case 403042:
+        return 'Invalid username or password';
+    }
+
+    return response.errorMessage;
+  }
+
+  mapSessionExpiration(rememberMe: boolean): number {
+    if (rememberMe) {
+      return -2;
```
Comment (bmmkoc) [inline src/app/core/gigya.service.ts]: Does -2 mean something? It's hard to read from the code. Maybe use a constant so it has a meaningful name?

File: src/app/core/gigya.service.ts (hunk 1 above)
Comment (bmxmaga) [inline src/app/core/gigya.service.ts]: Good point!
Gigya documentation specifies many error codes and I kept that in mind when writing this. But eventually, I consume only one of them, so you're right that this switch is pointless here :)

File: src/app/core/gigya.service.ts (hunk 2 above)
Comment (bmxmaga) [inline src/app/core/gigya.service.ts]: Yes, it's defined in the Gigya documentation. I'll put the link in the comment.

**Pull request merged**